File: LegalClaro-extension/create_icons.js
const fs = require('fs');
const path = require('path');

// Simple base64 encoded PNG icons (minimal legal scale icons)
const icons = {
  16: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAAAdgAAAHYBTnsmCAAAABl0RVh0U29mdHdhcmUAd3d3Lmlua3NjYXBlLm9yZ5vuPBoAAAE5SURBVDiNpZM9SwNBEIafRVsrwcJCG1sLG1sLbWwsrLSx0MZCGwttbCy0sdDGQhsLbSy0sdDGQhsLbSy0sdDGQhsLbSy0sRDEQhALQSxEsRDFQhQLUSxEsRDFQhQLUSy0sdDGQhsLbSy0sdDGQhsLbSy0sdDGQhsLbSy0sdDGQhsLbSy0sdDGQhsLbSy0sdDGQhsLbSy0sRDEQhALQSxEsRDFQhQLUSxEsRDFQhQLbSy0sdDGQhsLbSy0sdDGQhsLbSy0sdDGQhsLbSy0sdDGQhsLbSy0sdDGQhsLbSy0sRDEQhALQSxEsRDFQhQLUSxEsRDFQhQLUSy0sdDGQhsLbSy0sdDGQhsLbSy0sdDGQhsLbSy0sdDGQhsLbSxEsRDFQhQLUSxEsRDFQhALQSxEsRDFQhQLUSxEsRDFQhALQSxEsRDFQhQLUSxEsRDFQhQLQSw==',
  48: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADAAAAAwCAYAAABXAvmHAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAAOxAAADsQBlSsOGwAAABl0RVh0U29mdHdhcmUAd3d3Lmlua3NjYXBlLm9yZ5vuPBoAAANkSURBVGiB7ZnBaxNBFMafJCm0tYU2FtpYaGOhjYU2FtpYaGOhjYU2FmJjITYWYmMhNhZiYyE2FmJjITYWYmMhNhZiYyE2FmJjITYWYmMhNhZiYyE2FmJjITYWYmMhNhZiYyE2FmJjITYWYmMhNhZiYyE2FmJjITYWYmMhNhZiYyE2FmJjITYWYmMhNhZiYyE2FmJjITYWYmMhNhZiYyE2FmJjITYWYmMhNhZi',
  128: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAIAAAACACAYAAADDPmHLAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAAHYgAAB2IBOHqZ2wAAABl0RVh0U29mdHdhcmUAd3d3Lmlua3NjYXBlLm9yZ5vuPBoAAAlNSURBVHic7Z3Bb9tGFMc/kmzJlmTLcuLETpM4aZq2adu0TdN2Xdu1XTc0GwYMw4YNG4YNGzZs2LBhw4YNG4YN+wsYNmzYsGHDhg0bNmzYsGHDhg0bNmzYsGHDhg0b'
};

// Create icons directory
if (!fs.existsSync('icons')) {
  fs.mkdirSync('icons');
}

// Create simple colored square icons as placeholder
const sizes = [16, 48, 128];
sizes.forEach(size => {
  // Create a simple SVG icon
  const svg = `
    <svg width="${size}" height="${size}" xmlns="http://www.w3.org/2000/svg">
      <defs>
        <linearGradient id="grad" x1="0%" y1="0%" x2="100%" y2="100%">
          <stop offset="0%" style="stop-color:#667eea;stop-opacity:1" />
          <stop offset="100%" style="stop-color:#764ba2;stop-opacity:1" />
        </linearGradient>
      </defs>
      <rect width="${size}" height="${size}" fill="url(#grad)" />
      <text x="${size/2}" y="${size/2}" text-anchor="middle" dominant-baseline="middle" 
            fill="white" font-family="Arial" font-size="${size*0.7}">⚖</text>
    </svg>
  `;
  
  // Save as SVG first (can be converted to PNG later)
  fs.writeFileSync(path.join('icons', `icon${size}.svg`), svg);
  console.log(`Created icon${size}.svg`);
});

console.log('Icons created! Convert SVG to PNG using online tools or browser.');
console.log('For now, you can test the extension with SVG files.');

File: LegalClaro-extension/create_icons.py
import base64
from PIL import Image, ImageDraw, ImageFont
import os

# Create icons directory
os.makedirs('icons', exist_ok=True)

# Icon sizes
sizes = [16, 48, 128]

for size in sizes:
    # Create image with gradient background
    img = Image.new('RGBA', (size, size), (102, 126, 234, 255))
    draw = ImageDraw.Draw(img)
    
    # Create gradient effect
    for y in range(size):
        r = int(102 + (118-102) * y / size)
        g = int(126 + (75-126) * y / size) 
        b = int(234 + (162-234) * y / size)
        draw.line([(0, y), (size, y)], fill=(r, g, b, 255))
    
    # Draw scale emoji (using text approximation)
    try:
        font_size = int(size * 0.7)
        # Try to use system font with emoji support
        font = ImageFont.truetype("seguiemj.ttf", font_size)
    except:
        # Fallback to default font
        try:
            font = ImageFont.load_default()
        except:
            font = None
    
    if font:
        # Draw emoji
        text = "⚖"
        bbox = draw.textbbox((0, 0), text, font=font)
        w, h = bbox[2] - bbox[0], bbox[3] - bbox[1]
        x = (size - w) // 2
        y = (size - h) // 2
        draw.text((x, y), text, fill='white', font=font)
    else:
        # Fallback: draw simple geometric representation
        center = size // 2
        # Draw balance beam
        beam_width = int(size * 0.6)
        beam_height = max(2, size // 16)
        draw.rectangle([center - beam_width//2, center - beam_height//2, 
                       center + beam_width//2, center + beam_height//2], 
                      fill='white')
        # Draw center post
        post_width = max(2, size // 16)
        post_height = int(size * 0.4)
        draw.rectangle([center - post_width//2, center, 
                       center + post_width//2, center + post_height], 
                      fill='white')
    
    # Save PNG
    img.save(f'icons/icon{size}.png', 'PNG')
    print(f"Created icon{size}.png")